rewriter/ChainedSig.cc

```diff
@@ -0,0 +1,184 @@
+#include "rewriter/ChainedSig.h"
+#include "ast/Helpers.h"
+#include "ast/treemap/treemap.h"
+#include "core/errors/rewriter.h"
+
+using namespace std;
+namespace sorbet::rewriter {
+
+struct ChainedSigWalk {
+    // The previousSigSend is used to allow us to add errors to `sig` with missing block without accidentally adding it
+    // to a chained sig. E.g.:
+    // `sig` -> error no block
+    // `sig.final {}` -> no error. Need to prevent adding the error when we receive `sig` by itself
+    ast::Send *previousSigSend;
```

Comment:
Let's remove this and have `firstSendIsSig` return the relevant `Send*`, if any, rather than setting state shared between functions.  That way we can re-use that returned `Send*` later in the function, for determining whether we have a `final` method, I think?